src/mlpack/core/optimizers/cmaes/cmaes.hpp

```diff
@@ -0,0 +1,177 @@
+//! Minimal fitness value. Only activated if flg is true.
+struct { bool flg; double val; } stStopFitness;
+//! Minimal value difference.
+double stopTolFun;
+//! Minimal history value difference.
+double stopTolFunHist;
+//! Minimal search space step size.
+double stopTolX;
+//! Defines the maximal condition number.
+double stopTolUpXFactor;
+
+/* internal evolution strategy parameters */
+/**
+ * Population size. Number of samples per iteration, at least two,
+ * generally > 4.
+ */
+int lambda;
+/**
+ * Number of individuals used to recompute the mean.
+ */
+int mu;
+double mucov;
+/**
+ * Variance effective selection mass, should be lambda/4.
+ */
+double mueff;
+/**
+ * Weights used to recombinate the mean sum up to one.
+ */
+arma::vec weights;
+/**
+ * Damping parameter for step-size adaption, d = inifinity or 0 means adaption
+ * is turned off, usually close to one.
+ */
+double damps;
+/**
+ * cs^-1 (approx. n/3) is the backward time horizon for the evolution path
+ * ps and larger than one.
+ */
+double cs;
+double ccumcov;
+/**
+ * ccov^-1 (approx. n/4) is the backward time horizon for the evolution path
+ * pc and larger than one.
+ */
+double ccov;
+double diagonalCov;
+struct { double modulo; double maxtime; } updateCmode;
+double facupdateCmode;
+
+//! Step size.
+double sigma;
+//! Mean x vector, "parent".
+arma::vec xmean;
+//! Best sample ever.
+arma::vec xBestEver;
+//! x-vectors, lambda offspring.
+arma::mat population;
+//! Sorting index of sample population.
+arma::uvec index;
+//! History of function values.
+arma::vec funcValueHistory;
+
+double chiN;
+//! Lower triangular matrix: i>=j for C[i][j].
+arma::mat C;
+//! Matrix with normalize eigenvectors in columns.
+arma::mat B;
+//! Axis lengths.
+arma::vec rgD;
+//! Anisotropic evolution path (for covariance).
+arma::vec pc;
+//! Isotropic evolution path (for step length).
+arma::vec ps;
+//! Last mean.
+arma::vec xold;
+//! B*D*z.
+arma::vec BDz;
+//! Temporary (random) vector used in different places.
+arma::vec tempRandom;
+//! Objective function values of the population.
+arma::vec functionValues;
+//!< Public objective function value array returned by init().
+arma::vec publicFitness;
+
+//! Generation number.
+double gen;
+//! Algorithm state.
+enum {INITIALIZED, SAMPLED, UPDATED} state;
+
+// repeatedly used for output
+double maxdiagC;
+double mindiagC;
+double maxEW;
+double minEW;
+
+bool eigensysIsUptodate;
+bool doCheckEigen;
+double genOfEigensysUpdate;
+
+double dMaxSignifKond;
+
+double dLastMinEWgroesserNull;
+
+double countevals; //!< objective function evaluations
+
+void updateEigensystem(bool force);
+void sortIndex(const arma::vec rgFunVal, arma::vec& iindex, int n);
+void adaptC2(const int hsig);
+void addMutation(double* x, double eps = 1.0);
+
+void init(arma::vec& func);
+void samplePopulation();
+void updateDistribution(const arma::vec& fitnessValues);
+
+bool testForTermination();
+int  checkEigen(arma::vec diag, arma::mat Q);
```

Comment:
Can we remove the function, I think we use the return value from armadillo to determine if the eigen decomposition passed.